standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -498,20 +530,30 @@ public List<CompactionInfo> findReadyToCleanAborts(long abortedTimeThreshold, in
                     checkAbortedTimeThreshold && rs.getLong(4) + abortedTimeThreshold < systemTime;
             int numAbortedTxns = rs.getInt(5);
             if (numAbortedTxns > abortedThreshold || pastTimeThreshold) {
-              CompactionInfo info = new CompactionInfo();
+              AbortTxnRequestInfo info = new AbortTxnRequestInfo();
               info.dbname = rs.getString(1);
               info.tableName = rs.getString(2);
               info.partName = rs.getString(3);
               // In this case, this field contains min open write txn ID.
-              info.txnId = rs.getLong(6);
-              readyToCleanAborts.add(info);
+              info.minOpenWriteTxnId = rs.getLong(6);
```

Comment:
should we move `rs.getLong(6) > 0 ? rs.getLong(6) : Long.MAX_VALUE` logic here?

Reply:
Moved. Done.